### PR TITLE
Blog post

### PR DESCRIPTION
**Pull request by jfrolich**

### CODE REVIEW

File: documentation/docs/announcing-1.0.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+types. It's like having ES2040, but with the bad parts removed.
+
+When building a real app an important building block is data fetching. You need
+it for almost any kind of application. There are several ways to go about
+fetching data, but there is one protocol that's becoming more and more
+ubiquitous, and it looks like it has become the successor of the REST protocol.
+
+GraphQL is a great pairing with Reason, because the protocol is **typed**. This
+means that you have a predefined schema of data that your application can fetch.
+In priciple when we compose a query, given the schema, we know exactly what the
```
Comment (anmonteiro) [inline documentation/docs/announcing-1.0.md:18]:
```suggestion
In principle when we compose a query, given the schema, we know exactly what the
```

File: documentation/docs/announcing-1.0.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+This turned out to have major implications on the design and implementation of
+`graphql-ppx` itself, because to make this possible we needed to generate types.
+
+This rearchitecture in turn created a lot of opportunities to implement other
+things on the wishlist in a much better way. But if we would release these
+incrementally, we would have a number of breaking changes along the way. This
+would be a pretty bad experience for the users of the library and the authors of
+GraphQL clients. This is why we stayed in low-profile beta a little longer to
+ship something that is feature complete with no major breaking changes in the
+forseeable future.
```
Comment (anmonteiro) [inline documentation/docs/announcing-1.0.md:147]:
```suggestion
foreseeable future.
```

**Comment by jeddeloh** [comment]
This is great! If you’re still considering changes to the overall structure, what do you think about this?
- Condense the typescript part, remove it, or even make it a second article and link to it instead
- Lead with the language primitive idea (maybe there’s a way to hint at how much better this is than typescript approach in this short section and and have the same effect as before)
- Move right into the “What’s new” section that can wow people with tangible examples of benefits to the end user
- Then move into more advanced topics link then internals of graphql-ppx module structure, history, etc.

File: documentation/docs/announcing-1.0.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+This generates the following module (simplified):
+
+```reason
+module type UserQuery = {
+  module Raw: {
+    type t_user_type = string;
+    type t_user = {
+      id: string,
+      name: Js.Nullable.t(string),
+      type: t_user_type
```
Comment (jeddeloh) [inline documentation/docs/announcing-1.0.md:71]: Does this automatically become `type_` now?

File: documentation/docs/announcing-1.0.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    };
+    type t = {user: Js.Nullable.t(t_user)};
+  }
+  type t_user_type = [
+  | ADMIN
```
Comment (jeddeloh) [inline documentation/docs/announcing-1.0.md:76]:
```
| `ADMIN
| `MEMBER
```

File: documentation/docs/announcing-1.0.md (hunk 3 above)
Comment (jfrolich) [reply documentation/docs/announcing-1.0.md:71]: Yes but still a typo 🙈

File: documentation/docs/announcing-1.0.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+ubiquitous, and it looks like it has become the successor of the REST protocol.
+
+GraphQL is a great pairing with Reason, because the protocol is **typed**. This
+means that you have a predefined schema of data that your application can fetch.
+In principle when we compose a query, given the schema, we know exactly what the
+types of the data that is returned is going to be.
+
+In my experience the biggest source of bugs is that the data you get back is not
+what you expect. In TypeScript or JavaScript applications you might forget that
+a value can be `null` or you might even forgot to include a certain data
```
Comment (Zimmi48) [inline documentation/docs/announcing-1.0.md:23]:
```suggestion
a value can be `null` or you might even forget to include certain data
```

**Review by toto199 — APPROVED** [review]

**Comment by jfrolich** [comment]
This never happened :)

**Pull request closed without merging**